ReactiveCocoa/Swift/Property.swift

```diff
@@ -135,7 +128,7 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	/// followed by all changes over time, then complete when the property has
 	/// deinitialized.
 	public var producer: SignalProducer<Value, NoError> {
-		return SignalProducer { [getter, weak self] producerObserver, producerDisposable in
+		return SignalProducer { [_atomic, weak self] producerObserver, producerDisposable in
 			if let strongSelf = self {
```

Comment:
This could be simplified a bit if we grab the value via `withValue` first, and evaluate whether the property is deinitialized within the action. The locking is mandatory now anyway, so the comments should be removed too.